pkg/ccl/backupccl/restore_planning.go

```diff
@@ -646,6 +635,12 @@ func allocateDescriptorRewrites(
 				// Create the table rewrite with the new parent ID. We've done all the
 				// up-front validation that we can.
 				descriptorRewrites[table.ID] = &jobspb.RestoreDetails_DescriptorRewrite{ParentID: parentID}
+
+				if table.GetParentSchemaID() == keys.PublicSchemaIDForBackup ||
```

Comment:
nit: since you will be repushing, mind adding a comment here and below for future readers? This method is already complicated to understand as is.

Reply:
Done